File: core/src/main/java/com/nao4j/subtrol/core/document/internal/Price.java
package com.nao4j.subtrol.core.document.internal;

import lombok.Data;
import lombok.NonNull;

import java.math.BigDecimal;

import static java.math.BigDecimal.ZERO;

@Data
public class Price {

    @NonNull
    private final BigDecimal sum;
    @NonNull
    private final String currencyCode;

    public Price(final BigDecimal sum, final String currencyCode) {
        this.sum = sum;
        this.currencyCode = currencyCode;

        if (sum.compareTo(ZERO) < 0) {
            throw new IllegalArgumentException();   // todo: message
        }
    }

}

File: src/main/kotlin/com/nao4j/subtrol/model/Price.kt
package com.nao4j.subtrol.model

import java.math.BigDecimal
import java.math.BigDecimal.ZERO

data class Price(val sum: BigDecimal, val currencyCode: String) {

    init {
        if (sum <= ZERO) {
            throw IllegalArgumentException()
        }
    }

}
